src/pages/achievement/dashboard/subcomponents/AchievementModal.tsx

```diff
@@ -0,0 +1,87 @@
+import { Intent } from '@blueprintjs/core';
+import React from 'react';
+import { AchievementAbility, AchievementStatus } from 'src/commons/achievement/AchievementTypes';
+
+import AchievementModalCompletion from './modal/AchievementModalCompletion';
+import AchievementModalGoal from './modal/AchievementModalGoal';
+import AchievementWeek from './utils/AchievementWeek';
+import { prettifyDate } from './utils/DateHelper';
+import Inferencer from './utils/Inferencer';
+
+type AchievementModalProps = {
+  id: number;
+  inferencer: Inferencer;
+  handleGlow: any;
+};
+
+function AchievementModal(props: AchievementModalProps) {
+  const { id, inferencer, handleGlow } = props;
+
+  if (id < 0) return null;
+
+  const achievement = inferencer.getAchievementItem(id);
+  const { title, ability, deadline, goals, modal } = achievement;
+  const { modalImageUrl, description, completionText } = modal;
+
+  const status = inferencer.getStatus(id);
+  const awardedExp = inferencer.getStudentExp(id);
+
+  // Entire modal background
+  const handleBackground = (ability: AchievementAbility) => {
+    const assetUrl =
+      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/background';
```

Comment:
Seems like the game already has such a constant:

https://github.com/source-academy/cadet-frontend/blob/df9aebb8254ec254895405ebdc8cb5b386282b8d/src/features/game/commons/CommonConstants.ts#L4-L7

I think this can be extracted up to the common [Constants.ts](https://github.com/source-academy/cadet-frontend/blob/sa_2021_refactor/src/commons/utils/Constants.ts), and then the achievement asset URLs can be built off that.

Reply:
Done https://github.com/source-academy/cadet-frontend/commit/34d366e13e3fd9abda7c7e13376b84910fe0c3c8

Added the asset URL to the common [Constants.ts](https://github.com/source-academy/cadet-frontend/blob/34d366e13e3fd9abda7c7e13376b84910fe0c3c8/src/commons/utils/Constants.ts#L48)

https://github.com/source-academy/cadet-frontend/blob/34d366e13e3fd9abda7c7e13376b84910fe0c3c8/src/commons/utils/Constants.ts#L48

And achievement asset URL to the [AchievementConstants.ts](https://github.com/source-academy/cadet-frontend/blob/34d366e13e3fd9abda7c7e13376b84910fe0c3c8/src/commons/achievement/AchievementConstants.ts#L5)

https://github.com/source-academy/cadet-frontend/blob/34d366e13e3fd9abda7c7e13376b84910fe0c3c8/src/commons/achievement/AchievementConstants.ts#L5